test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -86,11 +87,14 @@ private void assertCommandBehavior(String inputCommand,
         if(isRelevantPersonsExpected){
             assertEquals(lastShownList, r.getRelevantPersons().get());
         }
-
+        //Parse the user command
+        Command command = new Parser().parseCommand(inputCommand);
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if (command.isMutating()) {
```

Comment:
How about the else branch? How would you go about the case where the command is not mutating?